api/src/pcapi/core/bookings/api.py

```diff
@@ -248,7 +245,7 @@ def book_offer(
         if is_activation_code_applicable:
             booking.activationCode = offers_repository.get_available_activation_code(stock)
             booking.mark_as_used()
-        if stock.price == 0 and stock.offer.subcategoryId in constants.FREE_OFFER_SUBCATEGORY_IDS_TO_ARCHIVE:
+        if stock.is_automatically_used():
```

Comment:
<!-- Issues represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 issue:** ‏My bad, il ne faut pas appeler la fonction mais la traiter comme une colonne
```suggestion
        if stock.is_automatically_used: